compiler/passes/convert-uast.cpp

```diff
@@ -2027,7 +2037,16 @@ struct Converter {
       ret = isCallExpr(expr) ? toCallExpr(expr) : new CallExpr(expr);
       addArgsTo = ret;
     } else {
-      ret = new CallExpr(convertAST(calledExpression));
+      Expr* baseExpr = convertAST(calledExpression);
+      if (SymExpr* se = toSymExpr(baseExpr)) {
+        if (isFnSymbol(se->symbol())) {
+          ret = new CallExpr(new UnresolvedSymExpr(se->symbol()->name));
```

Comment:
Can we put a comment saying this is a workaround? A reasonable alternative would be to adjust the Resolve.cpp code to bail out before establishing the type of a called function when there is just one candidate. Either way, I don't think we will want to generate UnresolvedSymExpr here in the long term, when we are using the dyno resolver to fully resolve calls.

Reply:
I just pushed a commit to adjust Resolver.cpp, which fixes the ``sparse.chpl`` failure. I also added a condition to the parenless-function case for good measure.